### PR TITLE
P3-261 Add tests infrastructure

### PR DESCRIPTION
**Pull request by afercia**
Please notice the typo in the branch name `P3-262-tests`. It should have been `P3-261-tests`
## Context

* We want to add infrastructure and configuration to run the tests.

## Summary

This PR can be summarized in the following changelog entry:

* Adds infrastructure and configuration to run unit tests.

## Relevant technical choices:

- This PR is based on the current Coding Standards implemented in https://github.com/Yoast/duplicate-post/pull/55 which don't use the YoastCS yet; the current plan is to add the YoastCS in the next weeks.
- Thus, I had to do a few things e.g.:
  - not use the array short syntax
  - disable some phpcs rules inline
  - meet the file / class name conventions of the WP standards, which differ from the YoastCS
- I added one simple test to verify the tests run correctly: worth reminding adding proper unit tests isn't in the scope of this PR
- **Important: I'd like to ask the Travis configuration to be reviewed by a specialist**: I made sure the tests run on Travis but this isn't my area of expertise so better to double check.

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

- run `composer install`
- run `composer check-cs` which was added in https://github.com/Yoast/duplicate-post/pull/55 and see there's only one warning related to pre-existing commented code
- run `composer test` and see the `test_duplicate_post_flatten_version` runs without errors
- optionally: alter the test data provider `flatten_version_provider()` in a way to make the test fail, run the test again and see it fails

### Test instructions for QA when the code is in the RC

* [ ] QA should use the same steps as above.

QA can test this PR by following these steps:

* N/A as it was decided the Rewrite & Republish feature should be tested as a whole once implementation is complete.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes

* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation

* [ ] I have written documentation for this change.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [x] I have added unittests to verify the code works as intended

Fixes [P3-261]

### CODE REVIEW

**Comment by afercia** [comment]
I think I forgot to add the Travis config to run the tests...

**Comment by afercia** [comment]
Added a very basic config to run the tests on Travis and made sure the tests run.

File: composer.json [hunk 1]
```diff
@@ -3,13 +3,20 @@
         "dealerdirect/phpcodesniffer-composer-installer": "^0.7.0",
         "wp-coding-standards/wpcs": "^2.1",
         "phpcompatibility/phpcompatibility-wp": "^2.1.0",
-        "automattic/vipwpcs": "^2.0"
+        "automattic/vipwpcs": "^2.0",
+        "phpunit/phpunit": "^5.7",
+        "brain/monkey": "^2.5"
```
Comment (karlijnbok) [inline composer.json:8]: Why 2.5? As in `wordpress-seo` it is 2.4?

File: phpunit.xml.dist [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		colors="true"
+		convertErrorsToExceptions="true"
+		convertNoticesToExceptions="true"
+		convertWarningsToExceptions="true"
+		processIsolation="false"
+		stopOnError="false"
+		stopOnFailure="false"
+		stopOnIncomplete="false"
+		stopOnSkipped="false"
+		verbose="true"
```
Comment (karlijnbok) [inline phpunit.xml.dist:17]: Was it intentional for these lines to be more indented than 
`xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"` and
`xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/7.5/phpunit.xsd"` ?

File: composer.json (hunk 1 above)
Comment (afercia) [reply composer.json:8]: Yep I noticed that. In `wpseo-news` is 2.5 though so I just used the most recent version we're using.

File: phpunit.xml.dist (hunk 2 above)
Comment (afercia) [reply phpunit.xml.dist:17]: Not intentional. Copy-pasted from `wpseo-news` 🙂 Please do feel free to fix the indentation.

**Comment by karlijnbok** [comment]
CR 👍

**Comment by enricobattocchi** [comment]
Acceptance test OK!

**Pull request merged**